### PR TITLE
fix(votes): stop showing raw html tags in vote cast drawer description

### LINKED ISSUE #2326: Vote description shows raw HTML tags in My Pending Actions vote drawer
## Problem

When a voter opens a pending vote from the My Pending Actions section of the Me dashboard, the vote description displays raw HTML markup (for example, visible `<p>` tags) as plain text instead of rendering the formatted content. This affects any vote whose description carries formatting — such as votes created in PCC with the rich text editor — and makes the voting panel look broken and harder to read.

## Outcome

The vote description in the pending-actions vote drawer renders as clean, readable content with its intended formatting. Voters never see raw HTML tags, and the description presentation is consistent with how the same vote appears in the vote results panel.

## Success criteria

- Opening a pending vote from My Pending Actions shows the description with no visible HTML tags
- Descriptions that contain formatting (paragraphs, line breaks) render readably
- Plain-text descriptions without formatting continue to display with their original line breaks

### PR DESCRIPTION
**Pull request by audigregorie**
## Summary

Vote descriptions authored in PCC with the rich-text editor carry HTML markup, but the My Pending Actions vote drawer rendered them as escaped plain text — voters saw literal `<p>` tags and the panel looked broken. This PR makes the vote cast drawer render the description's formatting, consistent with how the same vote already appears in the vote results panel.

Resolves #2326

## Behavior changes

| Before | After |
|---|---|
| Opening a pending vote from My Pending Actions showed the vote description as raw text with visible HTML tags (e.g. `<p>`) whenever the description was written with PCC's rich text editor | The description renders with its intended formatting — paragraphs and line breaks display cleanly, matching the vote results panel; plain-text descriptions without formatting continue to show their original line breaks |

## Technical changes

`apps/lfx-one/src/app/modules/votes/components/vote-cast-drawer/vote-cast-drawer.component.html` — Vote cast drawer template

- Replaced `{{ voteData.description }}` interpolation with an `[innerHTML]="voteData.description"` binding so rich-text descriptions render their markup instead of showing literal tags
- Kept `whitespace-pre-line` so legacy plain-text descriptions retain their line breaks (ticket acceptance criterion), and kept the `data-testid="vote-cast-drawer-description"` hook and the `@if (voteData.description)` guard
- Added two HTML comments recording the intent: the description is poll-creator-authored rich HTML, Angular's built-in `[innerHTML]` sanitization applies, and `whitespace-pre-line` is retained on purpose for plain-text content

**Security note:** the `[innerHTML]` binding relies on Angular's default HTML sanitizer (`SecurityContext.HTML` — strips `<script>`, inline event handlers, and `javascript:` URLs); there is no `bypassSecurityTrustHtml` anywhere in the votes module. The field is poll-creator-authored content from the PCC editor, and the identical raw binding for this exact field already ships on main in `vote-results-drawer.component.html` (6 usages). The `linkify` pipe was deliberately not applied: it escapes all markup, which would re-break the formatted descriptions this PR exists to render.

### CODE REVIEW

File: apps/lfx-one/src/app/modules/votes/components/vote-cast-drawer/vote-cast-drawer.component.html [hunk 1]
```diff
@@ -80,9 +80,9 @@ <h1 class="text-lg sm:text-xl font-semibold text-slate-900 flex-1 min-w-0 trunca
         @if (voteData.description) {
           <div class="flex flex-col gap-4" data-testid="vote-cast-drawer-details">
             <h2 class="text-lg font-semibold text-slate-900">Vote Details</h2>
-            <div class="text-sm text-slate-700 whitespace-pre-line" data-testid="vote-cast-drawer-description">
-              {{ voteData.description }}
-            </div>
+            <!-- Description is poll-creator-authored rich HTML (PCC editor); Angular's [innerHTML] sanitizer applies. Matches vote-results-drawer. -->
+            <!-- whitespace-pre-line is intentional: preserves line breaks in legacy plain-text descriptions (GH-2326). -->
+            <div class="text-sm text-slate-700 whitespace-pre-line" [innerHTML]="voteData.description" data-testid="vote-cast-drawer-description"></div>
```
[the remark this answers is not in the document]
Comment (audigregorie) [reply apps/lfx-one/src/app/modules/votes/components/vote-cast-drawer/vote-cast-drawer.component.html:85]: Fixed in dfdb2af3 — added a GH-2326 describe block to vote-cast-drawer.component.spec.ts: a rich-HTML description (`<p>First</p><p>Second</p>`) asserts real `<p>` elements render (and no literal tag text), and a newline-only plain-text description asserts `whitespace-pre-line` + intact newlines. Both fail against the pre-fix interpolation template.

**Review by MRashad26 — COMMENTED** [review]
Reviewed this diff plus ran the code-standards-enforcer agent (`~/LFX/code-enforcer-agent.md`) against this repo's actual `CLAUDE.md`/`.claude/rules/*.md`, then independently verified both surviving candidates. This is a small, well-scoped change — swapping an escaping `{{ }}` interpolation for `[innerHTML]` on `voteData.description`, matching an existing precedent already used 6x on `main` in `vote-results-drawer.component.html`. No Angular/CLAUDE.md convention violations found (standalone component, no functions in template, no rule specifically governs `[innerHTML]`/sanitization here and none is violated). Two minor findings, no correctness/security blockers.

**Test-coverage gap on the PR's own security claim (`vote-cast-drawer.component.spec.ts:158`).** The PR description claims Angular's default sanitizer strips `<script>`, inline event handlers, and `javascript:` URLs — but no added test exercises any attack-style payload, only benign `<p>` markup and plain text. A future refactor (e.g. switching to `bypassSecurityTrustHtml`, a custom pipe, or a sanitizer-default change on an Angular upgrade) could silently reintroduce XSS via `voteData.description` with zero test failure, since nothing currently asserts that malicious content actually gets neutralized. Worth adding a case with a `<script>`/`onerror`/`javascript:` payload that asserts it's stripped or escaped.

**Visual inconsistency with the sibling drawer (`vote-cast-drawer.component.html:287`).** This component keeps `whitespace-pre-line` alongside the new `[innerHTML]` binding, while `vote-results-drawer.component.html` renders the identical field via `[innerHTML]` without it — and neither component overrides Tailwind's preflight (which zeroes default `<p>` margins). If PCC's rich-text HTML for a description contains a literal newline/whitespace text node between block tags (common from pretty-printed or copy-pasted HTML, e.g. `<p>First</p>\n<p>Second</p>`), `white-space: pre-line` would render that newline as an extra visible line break here on top of the `<p>` block gap, while the same description in the results drawer wouldn't show that extra gap — an unintended visual difference between two views of the same vote. Worth reconciling (either drop `whitespace-pre-line` here to match, or add it to the results drawer too, whichever is the intended look).

Posting as Comment — neither finding blocks merge, but the missing attack-payload test is worth adding given the PR explicitly leans on sanitizer behavior as its security argument.

**Comment by audigregorie** [comment]
@MRashad26 Thanks for the thorough review — both findings verified against the current code.

**1. Test-coverage gap (accepted, fixed in `4b5102c`):** Added a `neutralizes attack payloads via the Angular sanitizer` test to the GH-2326 spec block. It feeds `<script>alert(1)</script>`, ``, and an `<a href="javascript:...">` payload through `[innerHTML]` and asserts the script element is gone, the event-handler attribute is stripped, and the URL is neutralized (Angular prefixes it `unsafe:`), while benign markup survives. Verified passing — any future regression (custom pipe, `bypassSecurityTrustHtml`) now fails loudly.

**2. `whitespace-pre-line` asymmetry (kept as-is, rationale):** The class is deliberate — legacy plain-text descriptions rely on it for line breaks, and that behavior is pinned by the `preserves a newline-only plain-text description` spec test plus the template comment on line 84. You are right that pretty-printed HTML with `\n` between block tags renders one extra break here vs. the results drawer; we accept that cosmetic edge case (PCC's editor emits compact HTML) rather than regress plain-text rendering or change the results drawer's 6 sibling renderings. If the team wants matching plain-text handling in the results drawer, that's a clean follow-up of its own.

**Review by MRashad26 — APPROVED** [review]
